<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>800</width>
    <height>600</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Anisotropic Kuwahara Filtering on the GPU</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QHBoxLayout" name="horizontalLayout_3">
    <item>
     <layout class="QVBoxLayout" name="verticalLayout">
      <item>
       <widget class="GLView" name="glw" native="true">
        <property name="sizePolicy">
         <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
          <horstretch>100</horstretch>
          <verstretch>100</verstretch>
         </sizepolicy>
        </property>
        <property name="minimumSize">
         <size>
          <width>500</width>
          <height>400</height>
         </size>
        </property>
       </widget>
      </item>
      <item>
       <layout class="QHBoxLayout" name="timeLayout">
        <property name="sizeConstraint">
         <enum>QLayout::SetDefaultConstraint</enum>
        </property>
        <item>
         <widget class="QToolButton" name="playButton">
          <property name="text">
           <string>&gt;</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QSlider" name="timeSlider">
          <property name="tracking">
           <bool>false</bool>
          </property>
          <property name="orientation">
           <enum>Qt::Horizontal</enum>
          </property>
         </widget>
        </item>
       </layout>
      </item>
     </layout>
    </item>
    <item>
     <layout class="QVBoxLayout" name="verticalLayout_2">
      <item>
       <widget class="QLabel" name="label">
        <property name="text">
         <string/>
        </property>
        <property name="pixmap">
         <pixmap resource="resources.qrc">:/logo.png</pixmap>
        </property>
        <property name="alignment">
         <set>Qt::AlignCenter</set>
        </property>
       </widget>
      </item>
      <item>
       <layout class="QFormLayout" name="formLayout">
        <property name="fieldGrowthPolicy">
         <enum>QFormLayout::AllNonFixedFieldsGrow</enum>
        </property>
        <property name="labelAlignment">
         <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
        </property>
        <property name="horizontalSpacing">
         <number>12</number>
        </property>
        <property name="verticalSpacing">
         <number>6</number>
        </property>
        <property name="topMargin">
         <number>12</number>
        </property>
        <item row="0" column="0">
         <widget class="QLabel" name="label_2">
          <property name="text">
           <string>view</string>
          </property>
         </widget>
        </item>
        <item row="0" column="1">
         <widget class="QComboBox" name="view">
          <property name="currentIndex">
           <number>1</number>
          </property>
          <item>
           <property name="text">
            <string>Original</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>Result</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>Flow Field</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>Anisotropy</string>
           </property>
          </item>
         </widget>
        </item>
        <item row="1" column="0">
         <widget class="QLabel" name="label_3">
          <property name="text">
           <string>algorithm</string>
          </property>
         </widget>
        </item>
        <item row="1" column="1">
         <widget class="QComboBox" name="algorithm">
          <property name="frame">
           <bool>true</bool>
          </property>
          <item>
           <property name="text">
            <string>Kuwahara</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>Generalized Kuwahara</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>Anisotropic Kuwahara (v1)</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>Anisotropic Kuwahara (v2)</string>
           </property>
          </item>
         </widget>
        </item>
        <item row="2" column="0">
         <widget class="QLabel" name="label_4">
          <property name="text">
           <string>sigma_t</string>
          </property>
         </widget>
        </item>
        <item row="2" column="1">
         <widget class="QDoubleSpinBox" name="sigma_t">
          <property name="enabled">
           <bool>false</bool>
          </property>
          <property name="keyboardTracking">
           <bool>false</bool>
          </property>
          <property name="minimum">
           <double>0.000000000000000</double>
          </property>
          <property name="maximum">
           <double>10.000000000000000</double>
          </property>
          <property name="singleStep">
           <double>0.100000000000000</double>
          </property>
          <property name="value">
           <double>2.000000000000000</double>
          </property>
         </widget>
        </item>
        <item row="3" column="0">
         <widget class="QLabel" name="label_5">
          <property name="text">
           <string>alpha</string>
          </property>
         </widget>
        </item>
        <item row="3" column="1">
         <widget class="QDoubleSpinBox" name="alpha">
          <property name="enabled">
           <bool>false</bool>
          </property>
          <property name="keyboardTracking">
           <bool>false</bool>
          </property>
          <property name="minimum">
           <double>1.000000000000000</double>
          </property>
          <property name="maximum">
           <double>999.990000000000009</double>
          </property>
          <property name="singleStep">
           <double>1.000000000000000</double>
          </property>
          <property name="value">
           <double>1.000000000000000</double>
          </property>
         </widget>
        </item>
        <item row="4" column="0">
         <widget class="QLabel" name="label_10">
          <property name="text">
           <string>N</string>
          </property>
         </widget>
        </item>
        <item row="4" column="1">
         <widget class="QSpinBox" name="N">
          <property name="enabled">
           <bool>false</bool>
          </property>
          <property name="keyboardTracking">
           <bool>false</bool>
          </property>
          <property name="minimum">
           <number>3</number>
          </property>
          <property name="maximum">
           <number>8</number>
          </property>
          <property name="value">
           <number>8</number>
          </property>
         </widget>
        </item>
        <item row="5" column="0">
         <widget class="QLabel" name="label_6">
          <property name="text">
           <string>radius</string>
          </property>
         </widget>
        </item>
        <item row="5" column="1">
         <widget class="QDoubleSpinBox" name="radius">
          <property name="enabled">
           <bool>true</bool>
          </property>
          <property name="keyboardTracking">
           <bool>false</bool>
          </property>
          <property name="minimum">
           <double>1.000000000000000</double>
          </property>
          <property name="maximum">
           <double>20.000000000000000</double>
          </property>
          <property name="singleStep">
           <double>1.000000000000000</double>
          </property>
          <property name="value">
           <double>3.000000000000000</double>
          </property>
         </widget>
        </item>
        <item row="6" column="0">
         <widget class="QLabel" name="label_8">
          <property name="text">
           <string>smoothing</string>
          </property>
         </widget>
        </item>
        <item row="7" column="1">
         <widget class="QLabel" name="kernel">
          <property name="enabled">
           <bool>false</bool>
          </property>
          <property name="sizePolicy">
           <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <property name="minimumSize">
           <size>
            <width>96</width>
            <height>96</height>
           </size>
          </property>
          <property name="maximumSize">
           <size>
            <width>96</width>
            <height>96</height>
           </size>
          </property>
          <property name="text">
           <string>n/A</string>
          </property>
          <property name="scaledContents">
           <bool>true</bool>
          </property>
         </widget>
        </item>
        <item row="8" column="0">
         <widget class="QLabel" name="label_9">
          <property name="text">
           <string>q</string>
          </property>
         </widget>
        </item>
        <item row="8" column="1">
         <widget class="QDoubleSpinBox" name="q">
          <property name="enabled">
           <bool>false</bool>
          </property>
          <property name="keyboardTracking">
           <bool>false</bool>
          </property>
          <property name="minimum">
           <double>1.000000000000000</double>
          </property>
          <property name="maximum">
           <double>16.000000000000000</double>
          </property>
          <property name="singleStep">
           <double>0.250000000000000</double>
          </property>
          <property name="value">
           <double>8.000000000000000</double>
          </property>
         </widget>
        </item>
        <item row="6" column="1">
         <widget class="QDoubleSpinBox" name="smoothing">
          <property name="enabled">
           <bool>false</bool>
          </property>
          <property name="keyboardTracking">
           <bool>false</bool>
          </property>
          <property name="suffix">
           <string>%</string>
          </property>
          <property name="maximum">
           <double>100.000000000000000</double>
          </property>
          <property name="singleStep">
           <double>5.000000000000000</double>
          </property>
          <property name="value">
           <double>33.329999999999998</double>
          </property>
         </widget>
        </item>
        <item row="9" column="1">
         <widget class="QCheckBox" name="renderFullQuad">
          <property name="text">
           <string>render full quad</string>
          </property>
         </widget>
        </item>
       </layout>
      </item>
      <item>
       <spacer name="verticalSpacer">
        <property name="orientation">
         <enum>Qt::Vertical</enum>
        </property>
        <property name="sizeHint" stdset="0">
         <size>
          <width>20</width>
          <height>40</height>
         </size>
        </property>
       </spacer>
      </item>
      <item>
       <widget class="QLabel" name="pf">
        <property name="text">
         <string/>
        </property>
        <property name="alignment">
         <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
        </property>
       </widget>
      </item>
     </layout>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menuBar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>800</width>
     <height>27</height>
    </rect>
   </property>
   <widget class="QMenu" name="menuFile">
    <property name="title">
     <string>File</string>
    </property>
    <addaction name="actionOpen"/>
    <addaction name="actionSave"/>
    <addaction name="actionRecord"/>
    <addaction name="separator"/>
    <addaction name="actionQuit"/>
   </widget>
   <widget class="QMenu" name="menuView">
    <property name="title">
     <string>View</string>
    </property>
    <addaction name="actionZoomIn"/>
    <addaction name="actionZoomOut"/>
    <addaction name="actionReset"/>
    <addaction name="separator"/>
    <addaction name="actionBuildLog"/>
   </widget>
   <widget class="QMenu" name="menuHelp">
    <property name="title">
     <string>Help</string>
    </property>
    <addaction name="actionAbout"/>
   </widget>
   <addaction name="menuFile"/>
   <addaction name="menuView"/>
   <addaction name="menuHelp"/>
  </widget>
  <action name="actionOpen">
   <property name="text">
    <string>Open...</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+O</string>
   </property>
   <property name="shortcutContext">
    <enum>Qt::ApplicationShortcut</enum>
   </property>
  </action>
  <action name="actionSave">
   <property name="text">
    <string>Save...</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+S</string>
   </property>
   <property name="shortcutContext">
    <enum>Qt::ApplicationShortcut</enum>
   </property>
  </action>
  <action name="actionRecord">
   <property name="enabled">
    <bool>false</bool>
   </property>
   <property name="text">
    <string>Record...</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+R</string>
   </property>
   <property name="shortcutContext">
    <enum>Qt::ApplicationShortcut</enum>
   </property>
  </action>
  <action name="actionZoomIn">
   <property name="text">
    <string>Zoom In</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+=</string>
   </property>
   <property name="shortcutContext">
    <enum>Qt::ApplicationShortcut</enum>
   </property>
  </action>
  <action name="actionZoomOut">
   <property name="text">
    <string>Zoom Out</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+-</string>
   </property>
   <property name="shortcutContext">
    <enum>Qt::ApplicationShortcut</enum>
   </property>
  </action>
  <action name="actionReset">
   <property name="text">
    <string>Reset</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+1</string>
   </property>
   <property name="shortcutContext">
    <enum>Qt::ApplicationShortcut</enum>
   </property>
  </action>
  <action name="actionAbout">
   <property name="text">
    <string>About</string>
   </property>
  </action>
  <action name="actionQuit">
   <property name="text">
    <string>Quit</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+Q</string>
   </property>
   <property name="shortcutContext">
    <enum>Qt::ApplicationShortcut</enum>
   </property>
  </action>
  <action name="actionOpenSettings">
   <property name="text">
    <string>Open Settings...</string>
   </property>
  </action>
  <action name="actionSaveSettings">
   <property name="text">
    <string>Save Settings...</string>
   </property>
  </action>
  <action name="actionBuildLog">
   <property name="text">
    <string>Build Log</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+L</string>
   </property>
  </action>
 </widget>
 <customwidgets>
  <customwidget>
   <class>GLView</class>
   <extends>QWidget</extends>
   <header>glview.h</header>
   <container>1</container>
   <slots>
    <slot>reset()</slot>
    <slot>zoomIn()</slot>
    <slot>zoomOut()</slot>
    <slot>updateKernel()</slot>
    <slot>process()</slot>
    <slot>buildLog()</slot>
   </slots>
  </customwidget>
 </customwidgets>
 <tabstops>
  <tabstop>view</tabstop>
  <tabstop>algorithm</tabstop>
  <tabstop>sigma_t</tabstop>
  <tabstop>alpha</tabstop>
  <tabstop>N</tabstop>
  <tabstop>radius</tabstop>
  <tabstop>smoothing</tabstop>
  <tabstop>q</tabstop>
  <tabstop>playButton</tabstop>
  <tabstop>timeSlider</tabstop>
 </tabstops>
 <resources>
  <include location="resources.qrc"/>
 </resources>
 <connections>
  <connection>
   <sender>actionQuit</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>close()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>419</x>
     <y>403</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionZoomIn</sender>
   <signal>triggered()</signal>
   <receiver>glw</receiver>
   <slot>zoomIn()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>259</x>
     <y>333</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionZoomOut</sender>
   <signal>triggered()</signal>
   <receiver>glw</receiver>
   <slot>zoomOut()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>259</x>
     <y>333</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionReset</sender>
   <signal>triggered()</signal>
   <receiver>glw</receiver>
   <slot>reset()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>259</x>
     <y>333</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>view</sender>
   <signal>currentIndexChanged(int)</signal>
   <receiver>glw</receiver>
   <slot>update()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>656</x>
     <y>136</y>
    </hint>
    <hint type="destinationlabel">
     <x>259</x>
     <y>302</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>sigma_t</sender>
   <signal>valueChanged(double)</signal>
   <receiver>glw</receiver>
   <slot>process()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>656</x>
     <y>188</y>
    </hint>
    <hint type="destinationlabel">
     <x>259</x>
     <y>302</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>alpha</sender>
   <signal>valueChanged(double)</signal>
   <receiver>glw</receiver>
   <slot>process()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>656</x>
     <y>214</y>
    </hint>
    <hint type="destinationlabel">
     <x>259</x>
     <y>302</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>N</sender>
   <signal>valueChanged(int)</signal>
   <receiver>glw</receiver>
   <slot>updateKernel()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>656</x>
     <y>240</y>
    </hint>
    <hint type="destinationlabel">
     <x>259</x>
     <y>302</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>radius</sender>
   <signal>valueChanged(double)</signal>
   <receiver>glw</receiver>
   <slot>updateKernel()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>656</x>
     <y>266</y>
    </hint>
    <hint type="destinationlabel">
     <x>259</x>
     <y>302</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>smoothing</sender>
   <signal>valueChanged(double)</signal>
   <receiver>glw</receiver>
   <slot>updateKernel()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>656</x>
     <y>292</y>
    </hint>
    <hint type="destinationlabel">
     <x>259</x>
     <y>302</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>q</sender>
   <signal>valueChanged(double)</signal>
   <receiver>glw</receiver>
   <slot>process()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>656</x>
     <y>388</y>
    </hint>
    <hint type="destinationlabel">
     <x>259</x>
     <y>302</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionBuildLog</sender>
   <signal>triggered()</signal>
   <receiver>glw</receiver>
   <slot>buildLog()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>259</x>
     <y>302</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>renderFullQuad</sender>
   <signal>toggled(bool)</signal>
   <receiver>glw</receiver>
   <slot>process()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>678</x>
     <y>474</y>
    </hint>
    <hint type="destinationlabel">
     <x>259</x>
     <y>303</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
